resources/js/beatmap-discussions/renderers.tsx

```diff
@@ -14,16 +14,27 @@ export function createRenderer(ElementType: React.ElementType) {
   };
 }
 
-export function linkRenderer(astProps: JSX.IntrinsicElements['a']) {
+export function linkRenderer(astProps: JSX.IntrinsicElements['a'], alwaysShortenUrls = true) {
   const props = propsFromHref(astProps.href);
   const href = safeReactMarkdownUrl(props.href);
 
+  const override = alwaysShortenUrls
+    || astProps.children instanceof Array
+      && astProps.children.length > 0
+      && astProps.children[0] === astProps.href;
```

Comment:
Is this supposed to make `[thing](https://..discussion.link..)` link says `thing`? It doesn't seem to work

Reply:
![image](https://github.com/ppy/osu-web/assets/1694544/4655a0d8-b3b7-4314-92c4-a51f2bbf117a)

works for me ?_?

Reply:
discussion page always uses the discussion id text; that was the question of maybe it shouldn't or the text should always be the discussion/post id when it's a discussion.

> (or maybe it should alway override like discussions? or discussions not always override? 🤔 )

Reply:
oh, I thought that was a bug. What was the reason it currently always overrides the text?

Reply:
also the original question doesn't really make sense to me. Why would chat have shortened link just for beatmapset discussion? How about never shorten it for chat? Or alternatively label it with `modding:` or something and then maybe we can do `user:` for user link etc. (although that seems to be another rabbit hole)

Reply:
I fell like the answer is that it was just simpler to overwrite the content  when it was implemented, instead of checking if it needed to be replaced, which is moot now since it needs to check 👀